src/mapping/PetRadialBasisFctMapping.hpp

```diff
@@ -770,24 +770,25 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshFirstRound()
       return; // Ranks not at the interface should never hold interface vertices
 
   // Tags all vertices that are inside otherMesh's bounding box, enlarged by the support radius
-  for(mesh::Vertex& v : filterMesh->vertices()) {
-    bool isInside = true;
+  auto rtree = mesh::rtree::getVertexRTree(filterMesh);
+  mesh::Mesh::VertexContainer insideVerts;
+  if (not _basisFunction.hasCompactSupport()) {
+      insideVerts = filterMesh->vertices();
+  }
+  else {
     #if PETSC_MAJOR >= 3 and PETSC_MINOR >= 8
-    if (_basisFunction.hasCompactSupport()) {
-      for (int d = 0; d < v.getDimensions(); d++) {
-        if (v.getCoords()[d] < otherMesh->getBoundingBox()[d].first - _basisFunction.getSupportRadius() or
-            v.getCoords()[d] > otherMesh->getBoundingBox()[d].second + _basisFunction.getSupportRadius() ) {
-          isInside = false;
-          break;
-        }
-      }
-    }
+      namespace bgi = boost::geometry::index;
+      rtree->query(bgi::within(otherMesh->getBoundingBox()),
+          boost::make_function_output_iterator([&filterMesh, &insideVerts](size_t idx){
+            insideVerts.push_back(&filterMesh->vertices()[idx]);
+          }));
     #else
       #warning "Mesh filtering deactivated, due to PETSc version < 3.8. \
 preCICE is fully functional, but performance for large cases is degraded."
     #endif
-    if (isInside)
-      v.tag();
+  }
+  for (auto& vert : insideVerts){
+    vert.tag();
```

Comment:
Idea:

Move that loop (`for vert in insideVerts: vert.tag()`) into `if (not _basisFunction.hasCompactSupport)` and use the `make_function_output_iterator` to directly invoke `vert.tag()`. Got the idea?

Reply:
Clever.